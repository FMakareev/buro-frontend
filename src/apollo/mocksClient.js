import faker from 'faker';
import setupClient from './helpers/apolloClientMock';
import schema from './schema.graphqls';
import { userDocumentList } from './graphql/query/userDocumentList';
import { userList } from './graphql/query/userList';
import { userItem } from './graphql/query/userItem';
import { notificationList } from './graphql/query/notificationList';
import { ROLE_BANK, ROLE_BUREAU, ROLE_CLIENT } from '../shared/roles';

const defaultMocks = {
  Query: () => ({
    userList,
    userItem,
    userEmailItem: (query, { email }) => {
      console.log(query, email);
      switch (email) {
        case 'client@test.com': {
          return {
            ...userItem(),
            email: 'client@test.com',
            role: ROLE_CLIENT,
          };
        }
        case 'bank@test.com': {
          return {
            ...userItem(),
            email: 'bank@test.com',
            role: ROLE_BANK,
          };
        }
        case 'bureau@test.com': {
          return {
            ...userItem(),
            email: 'bureau@test.com',
            role: ROLE_BUREAU,
          };
        }
        default: {
          throw Error(
            JSON.stringify({
              userEmailItem: null,
              errors: [
                {
                  message: 'GraphQL error: user not found',
                },
              ],
            }),
          );
        }
      }
    },
    userDocumentList,
    notificationList,
  }),
  Mutation: () => ({
    /**
     * @params {func} mutation - этот же запрос
     * @params {object} props - аргументы которые были переданы
     * */
    createUser: (mutation, props) => props,
    updateUser: (mutation, props) => props,
    userResetPassword: (mutation, props) => props,
    createNotification: (mutation, props) =>
      // для имитации запроса к серверу с рандомной задержкой и результатом.
      new Promise((resolve, reject) => {
        setTimeout(() => {
          faker.random.number(1)
            ? resolve(JSON.stringify({ data: { createNotification: props } }))
            : reject(
                JSON.stringify({
                  errors: [
                    {
                      message: 'error!',
                    },
                  ],
                }),
              );
        }, faker.random.number(2000));
      }),
    updateNotification: (mutation, props) =>
      // для имитации запроса к серверу с рандомной задержкой и результатом.
      new Promise((resolve, reject) => {
        setTimeout(() => {
          faker.random.number(1)
            ? resolve(JSON.stringify({ data: { updateNotification: props } }))
            : reject(
                JSON.stringify({
                  errors: [
                    {
                      message: 'error!',
                    },
                  ],
                }),
              );
        }, faker.random.number(2000));
      }),
    userPasswordRecovery: (mutation, props) =>
      // throw Error(JSON.stringify({
      //   userPasswordRecovery: null,
      //   errors: [
      //     {
      //       message:'Error!'
      //     }
      //   ]
      // }))
      props,
  }),
};

export const mocksClient = setupClient(defaultMocks, schema)();

export default mocksClient;
