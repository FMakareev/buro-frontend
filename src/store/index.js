/* global isBrowser */
import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';

import { composeWithDevTools } from 'redux-devtools-extension';
import { rootReducer } from './reducers/index';

const createStoreMiddleware = applyMiddleware(thunkMiddleware)(createStore);
const initialState = isBrowser ? window.PRELOADED_REDUX_STATE : {};
initialState.user = isBrowser ? JSON.parse(localStorage.getItem('user')) : null;

const Store = createStoreMiddleware(rootReducer, initialState, composeWithDevTools());

export { Store };

export default Store;
