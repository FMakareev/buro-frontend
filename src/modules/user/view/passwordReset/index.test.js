import React from 'react';
import renderer from 'react-test-renderer';
import { ApolloProvider } from 'react-apollo';
import mocksClient from "../../../../apollo/mocksClient";

import { BrowserRouter } from 'react-router-dom';
import {PasswordResetPage} from './index';
import {StyledThemeProvider} from "../../../../styles/StyledThemeProvider";
import {Provider as ProviderRedux} from 'react-redux';
import { Store } from '../../../../store';


test('PasswordResetPage: рендер без ошибок', () => {
  const output = renderer.create(<StyledThemeProvider>
    <ProviderRedux store={Store}>
      <BrowserRouter>
        <ApolloProvider client={mocksClient}>
          <PasswordResetPage/>
        </ApolloProvider>
      </BrowserRouter>
    </ProviderRedux>
  </StyledThemeProvider>);
  expect(output).toMatchSnapshot();
});

