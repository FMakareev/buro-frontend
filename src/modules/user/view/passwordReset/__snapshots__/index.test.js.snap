// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordResetPage: рендер без ошибок 1`] = `
<div
  className="sc-bxivhb dUiVDW"
>
  <div
    className="sc-1jxwge6-0 sc-14k9gvj-0 sc-10559ym-0 bWdFnk"
  >
    <div
      className="sc-1jxwge6-0 wterhv-0 kZLvgz"
    >
      <div
        className="sc-1jxwge6-0 gHyFgL"
      >
        <svg
          fill="none"
          height="2.5625em"
          viewBox="0 0 36 41"
          width="2.25em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 40.0241C17.0435 40.0241 16.087 39.762 15.2174 39.2815L2.78261 32.0739C1.04348 31.0692 0 29.2345 0 27.2251V12.8099C0 10.8005 1.08695 8.96584 2.78261 7.96114L15.2174 0.753524C16.9565 -0.251175 19.087 -0.251175 20.7826 0.753524L33.2174 7.96114C34.9565 8.96584 36 10.8005 36 12.8099V27.2251C36 29.2345 34.913 31.0692 33.2174 32.0739L20.7826 39.2815C19.913 39.762 18.9565 40.0241 18 40.0241ZM18 0.884573C17.1739 0.884573 16.3913 1.10299 15.6522 1.49613L3.21739 8.74743C1.78261 9.5774 0.869565 11.15 0.869565 12.8099V27.2251C0.869565 28.8851 1.78261 30.4577 3.21739 31.2876L15.6522 38.4952C17.087 39.3252 18.8696 39.3252 20.3478 38.4952L32.7826 31.2876C34.2174 30.4577 35.1304 28.8851 35.1304 27.2251V12.8099C35.1304 11.15 34.2174 9.5774 32.7826 8.74743L20.3478 1.53981C19.6087 1.10299 18.7826 0.884573 18 0.884573Z"
            fill="url(#paint0_linear)"
          />
          <path
            d="M18.1304 10.6694C18.5217 10.6694 18.8695 10.8004 19.2173 10.9752L25.1304 14.4261C25.5652 14.6882 25.8695 15.0376 26.0869 15.4308L20.1304 4.29174C18.826 3.50545 17.1739 3.50545 15.8695 4.29174L5.52167 10.2763L18.1304 10.6694Z"
            fill="url(#paint1_linear)"
          />
          <path
            d="M10.0427 15.2124C10.2601 14.9066 10.5209 14.6008 10.8688 14.4261L16.7818 10.9752C17.2166 10.7131 17.6514 10.6257 18.1296 10.6694L5.52094 10.2762C4.21659 11.0625 3.3905 12.4604 3.3905 13.9893V25.9583L10.0427 15.2124Z"
            fill="url(#paint2_linear)"
          />
          <path
            d="M9.86946 24.5605C9.69554 24.211 9.6086 23.8616 9.6086 23.4684V16.6103C9.6086 16.1298 9.73903 15.6492 9.9999 15.2561L3.34772 26.0457C3.34772 27.5746 4.17382 28.9724 5.47816 29.7587L15.826 35.7432L9.86946 24.5605Z"
            fill="url(#paint3_linear)"
          />
          <path
            d="M17.8695 29.3218C17.4782 29.3218 17.1304 29.1908 16.7825 29.016L10.8695 25.5651C10.4347 25.303 10.1304 24.9536 9.91296 24.5604L15.8695 35.6995C17.1738 36.4858 18.826 36.4858 20.1304 35.6995L30.4782 29.715L17.8695 29.3218Z"
            fill="url(#paint4_linear)"
          />
          <path
            d="M25.9565 24.7789C25.7391 25.0846 25.4783 25.3904 25.1304 25.5651L19.2174 29.0161C18.7826 29.2782 18.3478 29.3655 17.8696 29.3218L30.4783 29.715C31.7826 28.9287 32.6087 27.5309 32.6087 26.002V14.033L25.9565 24.7789Z"
            fill="url(#paint5_linear)"
          />
          <path
            d="M26.0869 15.4745C26.2608 15.8239 26.3478 16.1734 26.3478 16.5665V23.4247C26.3478 23.9052 26.2173 24.3857 25.9565 24.7789L32.6086 13.9893C32.6086 12.4604 31.7826 11.0625 30.4782 10.2763L20.1304 4.29175L26.0869 15.4745Z"
            fill="url(#paint6_linear)"
          />
          <defs>
            <linearGradient
              gradientUnits="userSpaceOnUse"
              id="paint0_linear"
              x1="17.9888"
              x2="17.9888"
              y1="0.683894"
              y2="40.3346"
            >
              <stop
                stopColor="#61C3E2"
              />
              <stop
                offset="0.2048"
                stopColor="#61C3E2"
              />
              <stop
                offset="0.876"
                stopColor="#134177"
              />
            </linearGradient>
            <linearGradient
              gradientUnits="userSpaceOnUse"
              id="paint1_linear"
              x1="7.61302"
              x2="26.4297"
              y1="4.79977"
              y2="13.7152"
            >
              <stop
                stopColor="#61C3E2"
              />
              <stop
                offset="0.2048"
                stopColor="#61C3E2"
              />
              <stop
                offset="0.876"
                stopColor="#134177"
              />
            </linearGradient>
            <linearGradient
              gradientUnits="userSpaceOnUse"
              id="paint2_linear"
              x1="-0.314949"
              x2="16.8723"
              y1="21.4428"
              y2="9.66638"
            >
              <stop
                stopColor="#61C3E2"
              />
              <stop
                offset="0.2048"
                stopColor="#61C3E2"
              />
              <stop
                offset="0.876"
                stopColor="#134177"
              />
            </linearGradient>
            <linearGradient
              gradientUnits="userSpaceOnUse"
              id="paint3_linear"
              x1="10.0478"
              x2="8.38388"
              y1="36.643"
              y2="15.8088"
            >
              <stop
                stopColor="#61C3E2"
              />
              <stop
                offset="0.2048"
                stopColor="#61C3E2"
              />
              <stop
                offset="0.876"
                stopColor="#134177"
              />
            </linearGradient>
            <linearGradient
              gradientUnits="userSpaceOnUse"
              id="paint4_linear"
              x1="28.3647"
              x2="9.54796"
              y1="35.2057"
              y2="26.2903"
            >
              <stop
                stopColor="#61C3E2"
              />
              <stop
                offset="0.2048"
                stopColor="#61C3E2"
              />
              <stop
                offset="0.876"
                stopColor="#134177"
              />
            </linearGradient>
            <linearGradient
              gradientUnits="userSpaceOnUse"
              id="paint5_linear"
              x1="36.301"
              x2="19.1137"
              y1="18.5895"
              y2="30.3659"
            >
              <stop
                stopColor="#61C3E2"
              />
              <stop
                offset="0.2048"
                stopColor="#61C3E2"
              />
              <stop
                offset="0.876"
                stopColor="#134177"
              />
            </linearGradient>
            <linearGradient
              gradientUnits="userSpaceOnUse"
              id="paint6_linear"
              x1="25.9113"
              x2="27.5753"
              y1="3.3837"
              y2="24.218"
            >
              <stop
                stopColor="#61C3E2"
              />
              <stop
                offset="0.2048"
                stopColor="#61C3E2"
              />
              <stop
                offset="0.876"
                stopColor="#134177"
              />
            </linearGradient>
          </defs>
        </svg>
      </div>
      <h2
        className="sc-1jxwge6-0 sc-18ch56c-0 cTSlNq"
        color="color1"
        fontFamily="bold"
        fontSize="24px"
      >
        Reset your password
      </h2>
    </div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="sc-1jxwge6-0 wterhv-0 ipWEwI"
        width="100%"
      >
        <div
          className="sc-1jxwge6-0 kWWhlf"
          width="100%"
        >
          <div
            className="sc-1tlq1yi-0 dLPklW"
          >
            <div
              className="sc-1jxwge6-0 wterhv-0 cGlObv"
            >
              <div
                className="sc-1jxwge6-0 sc-1tlq1yi-1 bjrTHb"
                width="100%"
              >
                <input
                  aria-required={false}
                  className="sc-1tlq1yi-2 fRLrXW wsrc07-0 kZIRDH"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  placeholder="Email address"
                  type="email"
                  value=""
                />
                <div
                  className="sc-1tlq1yi-3 ktWGUq"
                >
                  <div
                    className="sc-1jxwge6-0 sc-18ch56c-0 kapYeH"
                    color="color1"
                    fill="inherit"
                    fontSize={11}
                  >
                    <svg
                      fill="inherit"
                      height="1em"
                      stroke="none"
                      viewBox="0 0 32 32"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <mask
                        fill="white"
                        id="path-1-inside-1"
                        stroke="none"
                      >
                        <path
                          d="M16 32C24.8365 32 32 24.8365 32 16C32 7.16345 24.8365 0 16 0C7.1634 0 0 7.16345 0 16C0 24.8366 7.1634 32 16 32ZM16 31C7.72895 31 1 24.271 1 16C1 7.729 7.72895 1 16 1C24.2711 1 31 7.729 31 16C31 24.271 24.271 31 16 31Z"
                          stroke="none"
                        />
                      </mask>
                      <path
                        d="M16 32C24.8365 32 32 24.8365 32 16C32 7.16345 24.8365 0 16 0C7.1634 0 0 7.16345 0 16C0 24.8366 7.1634 32 16 32ZM16 31C7.72895 31 1 24.271 1 16C1 7.729 7.72895 1 16 1C24.2711 1 31 7.729 31 16C31 24.271 24.271 31 16 31Z"
                        fill="inherit"
                        stroke="none"
                      />
                      <path
                        d="M16 33C25.3888 33 33 25.3888 33 16H31C31 24.2843 24.2843 31 16 31V33ZM33 16C33 6.61117 25.3888 -1 16 -1V1C24.2843 1 31 7.71574 31 16H33ZM16 -1C6.61111 -1 -1 6.61117 -1 16H1C1 7.71573 7.71569 1 16 1V-1ZM-1 16C-1 25.3888 6.61111 33 16 33V31C7.71569 31 1 24.2843 1 16H-1ZM16 30C8.28123 30 2 23.7188 2 16H0C0 24.8233 7.17666 32 16 32V30ZM2 16C2 8.28128 8.28124 2 16 2V0C7.17666 0 0 7.17672 0 16H2ZM16 2C23.7188 2 30 8.28128 30 16H32C32 7.17672 24.8233 0 16 0V2ZM30 16C30 23.7188 23.7188 30 16 30V32C24.8233 32 32 24.8233 32 16H30Z"
                        fill="inherit"
                        mask="url(#path-1-inside-1)"
                        stroke="none"
                      />
                      <mask
                        fill="black"
                        height="14"
                        id="path-3-outside-2"
                        maskUnits="userSpaceOnUse"
                        stroke="none"
                        width="14"
                        x="9"
                        y="9"
                      >
                        <rect
                          fill="white"
                          height="14"
                          stroke="none"
                          width="14"
                          x="9"
                          y="9"
                        />
                        <path
                          d="M16 10C11.0779 10 10 13.9077 10 15.4135C10 19.5993 12.0394 22 15.5958 22C18.5227 22 19.32 21.6103 19.4015 21.5662C19.5638 21.4773 19.6198 21.281 19.5319 21.122C19.444 20.963 19.2385 20.9024 19.0708 20.9835C19.0637 20.9871 18.3182 21.3355 15.5958 21.3355C11.3248 21.3355 10.6869 17.6254 10.6869 15.4135C10.6869 15.3662 10.7475 10.6645 16 10.6645C20.6229 10.6645 21.3131 13.6401 21.3131 15.4135C21.3131 18.6192 20.3563 18.99 19.6365 18.99C18.6049 18.99 18.3678 18.6737 18.3638 18.5406V17.8147C18.3659 17.7959 18.3666 17.7771 18.3661 17.7574C18.366 17.7543 18.3639 17.7515 18.3638 17.7484V13.0684C18.3638 12.8847 18.2102 12.7361 18.0203 12.7361C17.8305 12.7361 17.6769 12.8847 17.6769 13.0684L17.677 13.4344C17.3626 13.1633 16.6934 12.7361 15.5958 12.7361C13.8864 12.7361 12.8283 14.3701 12.8283 15.8837C12.8283 18.1266 13.7593 19.2639 15.5958 19.2639C16.6194 19.2639 17.354 18.7928 17.6769 18.5392C17.6742 18.6307 17.6912 18.8904 17.9338 19.1454C18.2551 19.4832 18.8279 19.6545 19.6365 19.6545C22 19.6545 22 16.5735 22 15.4135C22 13.7857 21.4154 10 16 10ZM15.5958 18.5994C14.1567 18.5994 13.5151 17.762 13.5151 15.8837C13.5151 14.8824 14.1765 13.4006 15.5958 13.4006C16.8076 13.4006 17.3214 14.0242 17.3409 14.0482C17.3492 14.0596 17.3586 14.0703 17.3687 14.0803C17.4653 14.1774 17.6769 14.4509 17.6769 14.6319L17.6773 14.6338L17.678 17.6498C17.6112 17.7134 17.4962 17.8138 17.3707 17.9174C17.3637 17.9242 16.6634 18.5994 15.5958 18.5994Z"
                          stroke="none"
                        />
                      </mask>
                      <path
                        d="M16 10C11.0779 10 10 13.9077 10 15.4135C10 19.5993 12.0394 22 15.5958 22C18.5227 22 19.32 21.6103 19.4015 21.5662C19.5638 21.4773 19.6198 21.281 19.5319 21.122C19.444 20.963 19.2385 20.9024 19.0708 20.9835C19.0637 20.9871 18.3182 21.3355 15.5958 21.3355C11.3248 21.3355 10.6869 17.6254 10.6869 15.4135C10.6869 15.3662 10.7475 10.6645 16 10.6645C20.6229 10.6645 21.3131 13.6401 21.3131 15.4135C21.3131 18.6192 20.3563 18.99 19.6365 18.99C18.6049 18.99 18.3678 18.6737 18.3638 18.5406V17.8147C18.3659 17.7959 18.3666 17.7771 18.3661 17.7574C18.366 17.7543 18.3639 17.7515 18.3638 17.7484V13.0684C18.3638 12.8847 18.2102 12.7361 18.0203 12.7361C17.8305 12.7361 17.6769 12.8847 17.6769 13.0684L17.677 13.4344C17.3626 13.1633 16.6934 12.7361 15.5958 12.7361C13.8864 12.7361 12.8283 14.3701 12.8283 15.8837C12.8283 18.1266 13.7593 19.2639 15.5958 19.2639C16.6194 19.2639 17.354 18.7928 17.6769 18.5392C17.6742 18.6307 17.6912 18.8904 17.9338 19.1454C18.2551 19.4832 18.8279 19.6545 19.6365 19.6545C22 19.6545 22 16.5735 22 15.4135C22 13.7857 21.4154 10 16 10ZM15.5958 18.5994C14.1567 18.5994 13.5151 17.762 13.5151 15.8837C13.5151 14.8824 14.1765 13.4006 15.5958 13.4006C16.8076 13.4006 17.3214 14.0242 17.3409 14.0482C17.3492 14.0596 17.3586 14.0703 17.3687 14.0803C17.4653 14.1774 17.6769 14.4509 17.6769 14.6319L17.6773 14.6338L17.678 17.6498C17.6112 17.7134 17.4962 17.8138 17.3707 17.9174C17.3637 17.9242 16.6634 18.5994 15.5958 18.5994Z"
                        fill="inherit"
                        stroke="none"
                      />
                      <path
                        d="M19.4015 21.5662L19.5205 21.7861L19.5216 21.7855L19.4015 21.5662ZM19.5319 21.122L19.7507 21.0011L19.7507 21.0011L19.5319 21.122ZM19.0708 20.9835L18.9619 20.7584L18.9581 20.7603L19.0708 20.9835ZM18.3638 18.5406H18.1137L18.1139 18.5482L18.3638 18.5406ZM18.3638 17.8147L18.1153 17.7869L18.1138 17.8007V17.8147H18.3638ZM18.3661 17.7574L18.1162 17.7629L18.1162 17.7631L18.3661 17.7574ZM18.3638 17.7484H18.1134L18.1142 17.7621L18.3638 17.7484ZM17.6769 13.0684L17.4269 13.0684L17.4269 13.0684L17.6769 13.0684ZM17.677 13.4344L17.5137 13.6237L17.9271 13.9802L17.927 13.4344L17.677 13.4344ZM17.6769 18.5392L17.9268 18.5466L17.9427 18.0127L17.5225 18.3426L17.6769 18.5392ZM17.9338 19.1454L17.7527 19.3177L17.7527 19.3177L17.9338 19.1454ZM22 15.4135L22.25 15.4135V15.4135L22 15.4135ZM17.3409 14.0482L17.5423 13.8997L17.535 13.8907L17.3409 14.0482ZM17.3687 14.0803L17.5458 13.9039L17.5455 13.9036L17.3687 14.0803ZM17.6769 14.6319H17.4269V14.6576L17.4321 14.6827L17.6769 14.6319ZM17.6773 14.6338L17.9273 14.6337L17.9273 14.6081L17.9221 14.583L17.6773 14.6338ZM17.678 17.6498L17.8503 17.8309L17.9281 17.757L17.928 17.6497L17.678 17.6498ZM17.3707 17.9174L17.2116 17.7245L17.2037 17.731L17.1964 17.7381L17.3707 17.9174ZM16 9.75C13.4622 9.75 11.888 10.7624 10.9551 11.9909C10.032 13.2066 9.75 14.6156 9.75 15.4135H10.25C10.25 14.7057 10.5069 13.4079 11.3533 12.2933C12.1899 11.1915 13.6157 10.25 16 10.25V9.75ZM9.75 15.4135C9.75 17.5437 10.2687 19.2554 11.2737 20.4383C12.2838 21.6272 13.7578 22.25 15.5958 22.25V21.75C13.8775 21.75 12.5536 21.1725 11.6548 20.1146C10.751 19.0508 10.25 17.4692 10.25 15.4135H9.75ZM15.5958 22.25C18.5365 22.25 19.384 21.86 19.5205 21.7861L19.2824 21.3464C19.2559 21.3607 18.509 21.75 15.5958 21.75V22.25ZM19.5216 21.7855C19.8049 21.6303 19.9058 21.2817 19.7507 21.0011L19.3131 21.243C19.3338 21.2804 19.3226 21.3244 19.2813 21.347L19.5216 21.7855ZM19.7507 21.0011C19.5948 20.719 19.2419 20.623 18.9619 20.7585L19.1796 21.2085C19.235 21.1818 19.2933 21.2071 19.3131 21.243L19.7507 21.0011ZM18.9581 20.7603C19.0041 20.7371 18.3062 21.0855 15.5958 21.0855V21.5855C18.3301 21.5855 19.1234 21.237 19.1834 21.2067L18.9581 20.7603ZM15.5958 21.0855C13.55 21.0855 12.4052 20.206 11.7563 19.0792C11.0955 17.9316 10.9369 16.5036 10.9369 15.4135H10.4369C10.4369 16.5353 10.5972 18.0682 11.3231 19.3287C12.0609 20.61 13.3706 21.5855 15.5958 21.5855V21.0855ZM10.9369 15.4135C10.9369 15.4152 10.9396 15.1288 11.0237 14.6996C11.1079 14.2705 11.2721 13.7057 11.5911 13.1448C12.22 12.0388 13.4669 10.9145 16 10.9145V10.4145C13.2806 10.4145 11.871 11.6411 11.1564 12.8977C10.4536 14.1337 10.4369 15.3716 10.4369 15.4135H10.9369ZM16 10.9145C18.2537 10.9145 19.5005 11.6366 20.1928 12.5274C20.8944 13.4301 21.0631 14.5523 21.0631 15.4135H21.5631C21.5631 14.5013 21.3868 13.2489 20.5876 12.2205C19.779 11.1802 18.3692 10.4145 16 10.4145V10.9145ZM21.0631 15.4135C21.0631 17.0026 20.8239 17.8348 20.5368 18.2652C20.397 18.4747 20.2459 18.589 20.1006 18.6536C19.9525 18.7194 19.7957 18.74 19.6365 18.74V19.24C19.8372 19.24 20.07 19.2143 20.3037 19.1105C20.5402 19.0053 20.7633 18.8265 20.9527 18.5427C21.324 17.9861 21.5631 17.0302 21.5631 15.4135H21.0631ZM19.6365 18.74C19.1422 18.74 18.8685 18.6636 18.7279 18.5923C18.5786 18.5165 18.6117 18.4691 18.6137 18.5331L18.1139 18.5482C18.1198 18.7453 18.2736 18.9225 18.5017 19.0382C18.7385 19.1583 19.0992 19.24 19.6365 19.24V18.74ZM18.6138 18.5406V17.8147H18.1138V18.5406H18.6138ZM18.6122 17.8424C18.6157 17.8111 18.6167 17.781 18.616 17.7517L18.1162 17.7631C18.1164 17.7732 18.116 17.7807 18.1153 17.7869L18.6122 17.8424ZM18.6161 17.752C18.6154 17.7241 18.6104 17.7019 18.6066 17.688C18.6046 17.6811 18.6027 17.6754 18.6015 17.6718C18.6003 17.6683 18.5992 17.6655 18.5991 17.6652C18.5986 17.664 18.5993 17.6658 18.6 17.6677C18.6008 17.67 18.6023 17.6742 18.6039 17.6797C18.6071 17.6906 18.612 17.7099 18.6134 17.7347L18.1142 17.7621C18.1156 17.7884 18.1208 17.8093 18.1245 17.8216C18.1263 17.8279 18.1281 17.8329 18.1291 17.8359C18.1297 17.8374 18.1301 17.8386 18.1304 17.8394C18.1306 17.8398 18.1307 17.8401 18.1308 17.8403C18.1308 17.8404 18.1308 17.8405 18.1308 17.8405C18.1309 17.8406 18.1309 17.8406 18.1309 17.8406C18.131 17.841 18.1302 17.8389 18.1293 17.8362C18.1283 17.8334 18.1266 17.8285 18.1249 17.8222C18.1215 17.8099 18.1167 17.7891 18.1162 17.7629L18.6161 17.752ZM18.6138 17.7484V13.0684H18.1138V17.7484H18.6138ZM18.6138 13.0684C18.6138 12.7389 18.3404 12.4861 18.0203 12.4861V12.9861C18.08 12.9861 18.1138 13.0305 18.1138 13.0684H18.6138ZM18.0203 12.4861C17.7003 12.4861 17.4269 12.7389 17.4269 13.0684H17.9269C17.9269 13.0305 17.9607 12.9861 18.0203 12.9861V12.4861ZM17.4269 13.0684L17.427 13.4344L17.927 13.4344L17.9269 13.0683L17.4269 13.0684ZM17.8403 13.2451C17.4897 12.9427 16.7644 12.4861 15.5958 12.4861V12.9861C16.6223 12.9861 17.2356 13.3839 17.5137 13.6237L17.8403 13.2451ZM15.5958 12.4861C13.6992 12.4861 12.5783 14.2869 12.5783 15.8837H13.0783C13.0783 14.4533 14.0736 12.9861 15.5958 12.9861V12.4861ZM12.5783 15.8837C12.5783 17.0337 12.816 17.9456 13.3299 18.5733C13.8515 19.2103 14.6234 19.5139 15.5958 19.5139V19.0139C14.7318 19.0139 14.1199 18.7488 13.7168 18.2565C13.306 17.7548 13.0783 16.9767 13.0783 15.8837H12.5783ZM15.5958 19.5139C16.6972 19.5139 17.4853 19.0075 17.8313 18.7358L17.5225 18.3426C17.2227 18.578 16.5416 19.0139 15.5958 19.0139V19.5139ZM17.4271 18.5318C17.423 18.6671 17.4504 19 17.7527 19.3177L18.1149 18.9731C17.9319 18.7808 17.9254 18.5943 17.9268 18.5466L17.4271 18.5318ZM17.7527 19.3177C18.1459 19.7311 18.8065 19.9045 19.6365 19.9045V19.4045C18.8494 19.4045 18.3643 19.2352 18.1149 18.9731L17.7527 19.3177ZM19.6365 19.9045C20.2875 19.9045 20.792 19.6901 21.1725 19.3386C21.5474 18.9925 21.787 18.5274 21.9425 18.0458C22.2518 17.0875 22.25 15.9885 22.25 15.4135H21.75C21.75 15.9985 21.7482 17.02 21.4667 17.8922C21.3267 18.3258 21.1232 18.7036 20.8333 18.9713C20.5491 19.2338 20.1673 19.4045 19.6365 19.4045V19.9045ZM22.25 15.4135C22.25 14.5773 22.1012 13.1623 21.2341 11.9508C20.3559 10.724 18.7733 9.75 16 9.75V10.25C18.6421 10.25 20.0595 11.1689 20.8275 12.2419C21.6065 13.3302 21.75 14.622 21.75 15.4135L22.25 15.4135ZM15.5958 18.3494C14.9268 18.3494 14.4928 18.1574 14.2143 17.7938C13.9241 17.4152 13.7651 16.8038 13.7651 15.8837H13.2651C13.2651 16.8419 13.4269 17.5883 13.8174 18.0979C14.2194 18.6227 14.8258 18.8494 15.5958 18.8494V18.3494ZM13.7651 15.8837C13.7651 15.4302 13.917 14.8598 14.2289 14.4089C14.5363 13.9643 14.9864 13.6506 15.5958 13.6506V13.1506C14.7859 13.1506 14.1957 13.5778 13.8176 14.1245C13.444 14.6648 13.2651 15.3359 13.2651 15.8837H13.7651ZM15.5958 13.6506C16.7089 13.6506 17.1578 14.2194 17.1468 14.2058L17.535 13.8907C17.485 13.8291 16.9062 13.1506 15.5958 13.1506V13.6506ZM17.1397 14.1966C17.1564 14.2193 17.1741 14.2394 17.1919 14.2571L17.5455 13.9036C17.5431 13.9012 17.5421 13.8999 17.5421 13.8998L17.1397 14.1966ZM17.1915 14.2568C17.2264 14.2917 17.292 14.3698 17.3475 14.4579C17.3748 14.5013 17.3965 14.542 17.4107 14.577C17.4177 14.5942 17.422 14.608 17.4245 14.6184C17.427 14.6289 17.4269 14.6331 17.4269 14.6319H17.9269C17.9269 14.4647 17.8379 14.2983 17.7705 14.1914C17.6961 14.0732 17.6076 13.966 17.5458 13.9039L17.1915 14.2568ZM17.4321 14.6827L17.4325 14.6845L17.9221 14.583L17.9217 14.5812L17.4321 14.6827ZM17.4273 14.6338L17.428 17.6498L17.928 17.6497L17.9273 14.6337L17.4273 14.6338ZM17.5058 17.4686C17.4449 17.5265 17.335 17.6227 17.2116 17.7245L17.5298 18.1102C17.6575 18.0048 17.7774 17.9002 17.8503 17.8309L17.5058 17.4686ZM17.1964 17.7381C17.202 17.7327 16.5626 18.3494 15.5958 18.3494V18.8494C16.7641 18.8494 17.5254 18.1157 17.545 18.0966L17.1964 17.7381Z"
                        fill="inherit"
                        mask="url(#path-3-outside-2)"
                        stroke="none"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-1jxwge6-0 hZdxoY"
          width="100%"
        >
          <button
            className="sc-72db93-0 bWXyHX mxzgdr-0 kHylNb"
            disabled={true}
            fontSize={6}
            size="medium"
            type="submit"
            width="100%"
          >
            <div
              className="sc-1jxwge6-0 wterhv-0 eZGOHC"
              width="100%"
            >
              <div />
              <div
                className="sc-1jxwge6-0 wterhv-0 irkFkV"
              >
                Send password reset mail
              </div>
              <div
                className="sc-1jxwge6-0 wterhv-0 irkFkV"
              >
                <div
                  className="sc-1jxwge6-0 sc-18ch56c-0 hgJjUO"
                  color="color1"
                  fontSize={12}
                >
                  <svg
                    fill="inherit"
                    height="32px"
                    viewBox="0 0 40 41"
                    width="32px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M33.5631 20.9259C33.6718 20.8075 33.7261 20.6563 33.7261 20.5051C33.7261 20.3557 33.6729 20.2058 33.5661 20.0873L26.1245 11.8052C25.8975 11.5524 25.5118 11.5344 25.2627 11.7655C25.0137 11.9961 24.9973 12.3884 25.2237 12.6406L31.6978 19.8463L6.875 19.8463L6.875 21.0994L31.7417 21.0994L25.1688 28.3019C24.9393 28.5517 24.9539 28.944 25.2017 29.1771C25.4482 29.4094 25.8346 29.3946 26.0635 29.1436L33.5631 20.9259Z"
                      fill="inherit"
                    />
                    <path
                      d="M20 40.8095C31.0457 40.8095 40 31.7131 40 20.4921C40 9.27104 31.0457 0.174591 20 0.174591C8.95425 0.174591 0 9.27104 0 20.4921C0 31.7131 8.95425 40.8095 20 40.8095ZM20 39.5397C9.66119 39.5397 1.25 30.995 1.25 20.4921C1.25 9.98919 9.66119 1.44443 20 1.44443C30.3388 1.44443 38.75 9.98919 38.75 20.4921C38.75 30.995 30.3388 39.5397 20 39.5397Z"
                      fill="inherit"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
